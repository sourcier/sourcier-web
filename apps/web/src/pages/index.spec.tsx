import React from 'react';
import renderer from 'react-test-renderer';
import * as Gatsby from 'gatsby';

import Index from './index';

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
useStaticQuery.mockImplementation(() => ({
  site: {
    siteMetadata: {
      title: 'title',
    },
  },
}));

beforeEach(() => {
  jest.clearAllMocks();
});

describe('Index', () => {
  it('should render successfully', () => {
    const tree = renderer.create(<Index />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
