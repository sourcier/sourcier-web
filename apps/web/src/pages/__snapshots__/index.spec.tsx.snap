// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should render successfully 1`] = `
Array [
  <nav
    className="shadow-sm bg-slate-900"
  >
    <div
      className="max-w-3xl px-4 mx-auto"
    >
      <div
        className="flex justify-between"
      >
        <div
          className="flex space-x-7"
        >
          <div>
            <a
              className="flex items-center px-2 py-4"
              href="/"
            >
              <span
                className="text-lg font-semibold text-white"
                data-testid="site-brand"
              />
            </a>
          </div>
          <div
            className="items-center hidden space-x-1 md:flex"
          >
            <a
              className="px-2 py-4 font-semibold text-white hover:text-pink-300"
              href="/about"
            >
              About
            </a>
            <a
              className="px-2 py-4 font-semibold text-white hover:text-pink-300"
              href="/blog"
            >
              Blog
            </a>
          </div>
        </div>
        <div
          className="flex items-center md:hidden"
        >
          <button
            className="text-pink-300 outline-none"
            onClick={[Function]}
          >
            <svg
              className="w-6 h-6"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      className="hidden"
    >
      <ul
        className="px-4"
      >
        <li>
          <a
            className="block px-2 py-4 text-sm text-white transition duration-300 active:bg-pink-300"
            href="/about"
          >
            About
          </a>
        </li>
        <li>
          <a
            className="block px-2 py-4 text-sm text-white transition duration-300 active:bg-pink-300"
            href="/blog"
          >
            Blog
          </a>
        </li>
      </ul>
    </div>
  </nav>,
  <main
    className="max-w-3xl px-4 py-8 mx-auto prose"
  >
    <article
      className="prose"
    >
      <p>
        I'm making this by following the Gatsby Tutorial.
      </p>
    </article>
  </main>,
]
`;
